src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
+            first = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> newNode = new Node<>(value, null, null);
+            Node<T> tempPoint = getByIndex(index);
+            newNode.prev = tempPoint.prev;
+            newNode.next = tempPoint;
+            if (tempPoint.prev != null) {
+                tempPoint.prev.next = newNode;
+            }
+            tempPoint.prev = newNode;
+            if (newNode.prev == null) {
+                first = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
+            for (T element : list) {
+                add(element);
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> tempPoint = getByIndex(index);
+        return tempPoint != null ? tempPoint.data : null;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T result = null;
+        Node<T> tempPoint = getByIndex(index);
+        if (tempPoint != null) {
+            result = tempPoint.data;
+            tempPoint.data = value;
+        }
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T result = null;
+        Node<T> tempPoint = getByIndex(index);
+        if (tempPoint != null) {
+            result = tempPoint.data;
+            if (tempPoint.next != null) {
+                tempPoint.next.prev = tempPoint.prev;
+            }
+            if (tempPoint.prev != null) {
+                tempPoint.prev.next = tempPoint.next;
+            }
+            if (tempPoint.prev == null) {
+                first = tempPoint.next;
+            }
+            if (tempPoint.next == null) {
+                last = tempPoint.prev;
+            }
+            size--;
+        }
+        return result;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        boolean result = false;
+        Node<T> tempPoint = first;
+        for (int i = 0; i < size; i++) {
+            if (tempPoint != null && (tempPoint.data == object || (tempPoint.data != null
+                    && tempPoint.data.equals(object)))) {
+                remove(i);
+                result = true;
+                break;
+            }
+            tempPoint = tempPoint.next;
+        }
+        return result;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getByIndex(int index) {
+        if (index < 0 || index > size - 1) {
+            throw new IndexOutOfBoundsException("Invalid index : " + index + " Out bounds of List");
+        }
```

Comment:
U throw an exception in the case index out of bound, so you can remove these `if` in the top regarding to comments

Reply:
ok